### DISCUSSION TITLE
Regarding Moose installation on HPC Cluster

### DISCUSSION THREAD

**Discussion by rakeshm22** [idaholab/moose#29376] [Q&A Tools]
Hello everyone,
I am trying to install MOOSE on our HPC cluster. I followed all the instructions mentioned on the [MOOSE HPC Installation Guide](https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html).

Here are the steps I followed:
```
(a) Currently Loaded Modules (as required for installing MOOSE on the HPC cluster):

1. python/conda-python/3.9
2. compiler/gcc/13.1.0
3. cmake/3.28.1
4. compiler/openmpi/gcc/4.0.2

(b) Activate Environment: Activated the desired MPI environment.
(c) Cloning MOOSE: Successfully cloned the MOOSE repository.
(d) PETSc, libMesh, and WASP:
```
Successfully ran the PETSc command.
However, while installing libMesh, I encountered the following error:
```
 *********************************************************************
                                        ERROR
********************************************************************
make[4]: Leaving directory `/home/rakeshm22/projects/moose/libmesh/build/contrib/netcdf/v4/libhdf5'
Making all in liblib
make[4]: Entering directory `/home/rakeshm22/projects/moose/libmesh/build/contrib/netcdf/v4/liblib'
  CC       libnetcdf_la-nc_initialize.lo
  CCLD     libnetcdf.la
/usr/bin/ld: cannot find -ludev
collect2: error: ld returned 1 exit status
make[4]: *** [libnetcdf.la] Error 1
make[4]: Leaving directory `/home/rakeshm22/projects/moose/libmesh/build/contrib/netcdf/v4/liblib'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/home/rakeshm22/projects/moose/libmesh/build/contrib/netcdf/v4'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/home/rakeshm22/projects/moose/libmesh/build/contrib/netcdf/v4'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/rakeshm22/projects/moose/libmesh/build/contrib'
make: *** [all-recursive] Error 1
**********************************************************************************************************
```

**Comment by GiudGiud** [comment]
Hello
```
/usr/bin/ld: cannot find -ludev
```
This is a missing dependency in your environment. 
you need to install it locally or get your cluster admins to make it available to you in another module
if you have admin rights and apt-get is installed (unlikely) you can just run 
```
sudo apt-get install --reinstall libudev-devel
```

**Reply by rakeshm22** [reply]
Hello GuidGiud,

Thank you for your response.

I am not a sudo user, so I kindly request your guidance on how to install it locally and resolve this issue. Your assistance would be greatly appreciated.

Thank you again for your help.

**Reply by GiudGiud** [reply]
I have never installed libudev from source. I m not sure how easy it is. 
can you see if you can find it in the modules?
```
module avail
```
should show everything you can load.
Your HPC team might also know if you they already have it with modules for different compiler stack

**Comment by rakeshm22** [comment]
Hello everyone,
I am trying to reproduce grain growth for a bicrystal system in the presence of a temperature gradient as an external field. I wrote a .i file based on the equations mentioned in the paper. 
I am not able to find any output. Kindly help in this regards.


 (https://github.com/user-attachments/assets/796e1709-6867-4548-955e-a7407dd1719a)
(https://github.com/user-attachments/files/18279534/Michael.R_2013.pdf)
#
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 20
  ny = 10
  xmin = 0
  xmax = 20
  ymin = 0
  ymax = 10
[]

[GlobalParams]
  block = 0
[]

[Variables]
  [./eta1]
  [../]
  
  [./eta2]
  [../]
  
  [./T]   
    order = FIRST
    family = LAGRANGE
  [../]
[]

[ICs]

 [./IC_eta1]
    x1 = 0
    y1 = 0
    x2 = 20
    y2 = 10
    inside = 1.0
    outside = 0.0
    type = BoundingBoxIC
    variable = eta1
    int_width = 0.5
  [../]

  [./IC_eta2]
    x1 = 10
    y1 = 0
    x2 = 20
    y2 = 10
    inside = 1.0
    outside = 0.0
    type = BoundingBoxIC
    variable = eta2
    int_width = 0.5
  [../]


   [./T]
    x1 = 0
    y1 = 0
    x2 = 18
    y2 = 10
    inside = 0.01
    outside = 0.01
    type = BoundingBoxIC
    variable = T
    int_width = 0.5
  [../]

[]


[AuxVariables]
  [./T_x]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./T_y]
    order = CONSTANT
    family = MONOMIAL
  [../]

  [./eta1_x]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eta1_y]
    order = CONSTANT
    family = MONOMIAL
  [../]

  [./eta2_x]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eta2_y]
    order = CONSTANT
    family = MONOMIAL
  [../]

[]



[AuxKernels]
  [./T_x]
    type = VariableGradientComponent
    variable = T_x
    component = x
    gradient_variable = T
  [../]
  [./T_y]
    type = VariableGradientComponent
    variable = T_y
    component = y
    gradient_variable = T
  [../]
  [./eta1_x]
    type =VariableGradientComponent
    variable = eta1_x
    component = x
    gradient_variable = eta1
  [../]

  [./eta1_y]
    type =VariableGradientComponent
    variable = eta1_y
    component = y
    gradient_variable = eta1
  [../]

  [./eta2_x]
    type =VariableGradientComponent
    variable = eta2_x
    component = x
    gradient_variable = eta2
  [../]

  [./eta2_y]
    type =VariableGradientComponent
    variable = eta2_y
    component = y
    gradient_variable = eta2
  [../]

[]


[Kernels]
 
  [./AC1_bulk]
    type = AllenCahn
    variable = eta1
    f_name = F
    args = 'eta2 T '
  [../]
  [./AC1_int]
    type = ACInterface
    variable = eta1
    kappa_name = kappa_s
  [../]
  [./e1_dot]
    type = TimeDerivative
    variable = eta1
  [../]
  
  [./AC2_bulk]
    type = AllenCahn
    variable = eta2
    f_name = F
    args = 'eta1 T '
  [../]
  [./AC2_int]
    type = ACInterface
    variable = eta2
    kappa_name = kappa_s
  [../]
  [./e2_dot]
    type = TimeDerivative
    variable = eta2
 [../]

[./diff2]
    type = Diffusion
    variable = T
  [../]

[]


[Materials]
 

 [./FreeEng]
    type = DerivativeParsedMaterial
    
    args = 'eta1 eta2 T_x T_y eta1_x eta1_y eta2_x eta2_y  T'
   
    f_name = F
    
    constant_names = ' ww'  
    constant_expressions = ' 1.14 '
   

  function = 'f2:=(1/4*eta1^4-1/2*eta1^2+1/4*eta2^4-1/2*eta2^2);
              f3:=s*(eta1^2*eta2^2);
              f4:=eta1*4.0*(T_x*eta1_x+T_y*eta1_y);  
              f5:=eta2*4.0*(T_x*eta2_x+T_y*eta2_y);

            F:= ww*(f2 + f3 + 0.25) + f4 + f5;F'


derivative_order = 2
 outputs = exodus
 
[../]


[./const]
    type = GenericConstantMaterial
    prop_names = 'kappa_s L '
    prop_values = '1.0 1.0 '
    outputs = exodus
  [../]

[]

[BCs]

  [./Periodic]
    [./all]
      auto_direction = 'y'
    [../]
  [../] 

[./neumaan1]
    type = NeumannBC
     boundary = 'left right'
     variable = eta1
     value = 0
  [../]
  [./neumaan2]
    type = NeumannBC
     boundary = 'left right'
     variable = eta2
     value = 0
  [../]

  [./Dirichlet1]
    type = DirichletBC
     boundary = 'left'
     variable = T
     value = 0.1
  [../]

  [./Dirichlet2]
    type = DirichletBC
     boundary = 'right'
     variable = T
     value = 1.5
  [../]

[]


[Postprocessors]
  [./dt]
    type = TimestepSize
  [../]


[]

[Preconditioning]
   active = ' '
  [./SMP]
    type = SMP
    full = true
    petsc_options_iname = '-ksp_gmres_restart -snes_atol -snes_rtol -ksp_rtol -pc_type  -build_twosided'
    petsc_options_value = '    160            1e-10      1e-8      1e-6       bjacobi       allreduce'
[../]
[]

[Executioner]
  type = Transient
  scheme = BDF2
  solve_type = newton

  petsc_options_iname = '-pc_type -ksp_grmres_restart -sub_ksp_type -sub_pc_type -pc_asm_overlap'
  petsc_options_value = 'asm      31                  preonly       lu           1'
  l_max_its = 30
  nl_abs_tol = 1e-7
  l_tol = 1e-7
  nl_max_its = 30
  nl_rel_tol = 1e-7
  end_time = 30000000
  dt=0.50



[]

[Outputs]
  execute_on = 'timestep_end'
  exodus = true
  interval=2
[]


#

(https://github.com/user-attachments/assets/796e1709-6867-4548-955e-a7407dd1719a)
(https://github.com/user-attachments/files/18279534/Michael.R_2013.pdf)

**Reply by GiudGiud** [reply]
hello 

please open a different post, the topic of this discussion is HPC installation not phase field
